internal/appsec/dyngo/instrumentation/httpinstr/tags.go

```diff
@@ -0,0 +1,91 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016 Datadog, Inc.
+
+package httpinstr
+
+import (
+	"encoding/json"
+	"sort"
+	"strings"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext"
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/log"
+)
+
+// setEventSpanTags sets the security event span tags into the service entry span.
+func setEventSpanTags(span ddtrace.Span, events json.RawMessage) {
+	// Set the appsec event span tag
+	// eventTag is the structure to use in the `_dd.appsec.json` span tag.
+	type eventTag struct {
+		Triggers json.RawMessage `json:"triggers"`
+	}
+	// TODO(Julio-Guerra): avoid serializing the json in the request hot path
+	event, err := json.Marshal(eventTag{Triggers: events})
+	if err != nil {
+		log.Error("appsec: unexpected error while serializing the appsec event span tag: %v", err)
+		return
+	}
+	span.SetTag("_dd.appsec.json", string(event))
+	// Keep this span due to the security event
+	span.SetTag(ext.ManualKeep, true)
+	span.SetTag("_dd.origin", "appsec")
+	// Set the appsec.event tag needed by the appsec backend
+	span.SetTag("appsec.event", true)
+}
+
+// setEventSpanTags sets the AppSec-specific span tags when a security event occurred into the service entry span.
+func setSpanTags(span ddtrace.Span, events json.RawMessage, remoteIP string, headers map[string][]string) {
+	setEventSpanTags(span, events)
+	span.SetTag("network.client.ip", remoteIP)
+	for h, v := range normalizeHTTPHeaders(headers) {
+		span.SetTag("http.request.headers."+h, v)
```

Comment:
What about sensitive information such as tokens and other things? Do we really want to set all of them here?

Reply:
We take that into account and only a given passlist of headers is in `headers`. You can find it at https://github.com/DataDog/dd-trace-go/pull/1068/files#diff-c7d5dea384b594ae273865d89ec68a573cdeb29c13c82010b30f5297bd0ae872R48